### PR TITLE
add support for Google Bare Metal Solution

### PR DESCRIPTION
**Pull request by alex-basinov**
Adds support for [BMS machines](https://cloud.google.com/bare-metal).

### CODE REVIEW

File: internal/resourcemapping/resourcemapping.go [hunk 1]
```diff
@@ -122,6 +123,10 @@ var (
 			},
 			awsAccount: {otelKeys: []string{string(semconv.CloudAccountIDKey)}},
 		},
+		bmsInstance: {
+			location:   {otelKeys: []string{string(semconv.CloudRegionKey)}},
```
Comment (aabmass) [inline internal/resourcemapping/resourcemapping.go:127]: Could location use the zone as well? Many others resources check for zone then region

**Review by aabmass — COMMENTED** [review]
@alex-basinov is this monitored resource publicly writeable yet? If it is, would you mind adding an integration test for it following https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/integrationtests.md

Lmk if you have any questions.

**Comment by alex-basinov** [comment]
> @alex-basinov is this monitored resource publicly writeable yet?

No, not yet but It'll become publicly writable very soon.

**Comment by alex-basinov** [comment]
The monitored resource is now publicly writeable. I've also added an integration test. PTAL ?

**Review by aabmass — APPROVED** [review]

**Pull request merged**